src/it/scala/com/exasol/cloudetl/BaseS3IntegrationTest.scala

```diff
@@ -0,0 +1,84 @@
+package com.exasol.cloudetl
+
+import java.io.File
+
+import com.exasol.dbbuilder.dialects.Table
+
+import com.amazonaws.services.s3.model._
+import com.amazonaws.services.s3.AmazonS3
+import com.amazonaws.services.s3.AmazonS3ClientBuilder
+import org.apache.hadoop.fs.{Path => HPath}
+import org.testcontainers.utility.DockerImageName
+import org.testcontainers.containers.localstack.LocalStackContainer
+import org.testcontainers.containers.localstack.LocalStackContainer.Service.S3
+
+trait BaseS3IntegrationTest extends BaseIntegrationTest {
+
+  val LOCALSTACK_DOCKER_IMAGE = DockerImageName.parse("localstack/localstack:0.12.5")
+  val s3Container = new LocalStackContainer(LOCALSTACK_DOCKER_IMAGE)
+    .withServices(S3)
+    .withReuse(true)
+  var s3: AmazonS3 = _
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    s3Container.start()
+    prepareS3Client()
+  }
+
+  override def afterAll(): Unit = {
+    s3Container.stop()
+    super.afterAll()
+  }
+
+  def prepareS3Client(): Unit =
+    s3 = AmazonS3ClientBuilder
+      .standard()
+      .withPathStyleAccessEnabled(true)
+      .withEndpointConfiguration(s3Container.getEndpointConfiguration(S3))
+      .withCredentials(s3Container.getDefaultCredentialsProvider())
+      .disableChunkedEncoding()
+      .build()
+
+  def createS3ConnectionObject(): Unit = {
+    val credentials = s3Container.getDefaultCredentialsProvider().getCredentials()
+    val awsAccessKey = credentials.getAWSAccessKeyId()
+    val awsSecretKey = credentials.getAWSSecretKey()
+    val secret = s"S3_ACCESS_KEY=$awsAccessKey;S3_SECRET_KEY=$awsSecretKey"
+    factory.createConnectionDefinition("S3_CONNECTION", "", "dummy_user", secret)
+    ()
+  }
+
+  def uploadFileToS3(bucket: String, file: HPath): Unit = {
+    s3.createBucket(new CreateBucketRequest(bucket))
+    val request = new PutObjectRequest(bucket, file.getName(), new File(file.toUri()))
+    s3.putObject(request)
+    Thread.sleep(3 * 1000)
+    ()
+  }
+
+  def importFromS3IntoExasol(
+    schemaName: String,
+    table: Table,
+    bucket: String,
+    file: String,
+    dataFormat: String
+  ): Unit = {
+    val s3Endpoint = s3Container
+      .getEndpointConfiguration(S3)
+      .getServiceEndpoint()
+      .replaceAll("127.0.0.1", "172.17.0.1")
```

Comment:
using hard-coded IP is dangerous. It only works by accident on most machines. You can also configure docker to use a different network.

Solutions:

* user a docker network + dns
* map ports to the host and use `EXASOL_CONTAINER.getHostIp()`

Reply:
Yes, all correct.

Here do not use the IPv4 address, but the docker bridge (172.17.0.0/16) gateway address. With network we would still require this replacement because the S3 returns localhost (127.0.0.1) for its service endpoint.

Reply:
sure, but also docker network can have different IPs.
With docker network you could replace with the name of the container.

Reply:
Yes.

But it does not work out of box for GenericContainer. I have still to update the `/etc/hosts` file in Exasol container, since it could not resolve the integrated container name.

Reply:
Here I update the hosts file so it can find the worker container: https://github.com/exasol/cloud-storage-extension/pull/139/files#diff-33e9014839a04c627eb21a81d3e06ed2fe0ee50c07b56ab09a09a61470a507cbR96-R100